src/bin/NodeOptions.ts

```diff
@@ -0,0 +1,128 @@
+  public websocketPort: number;
+  /** If set to true, the container will not be deleted when it is stopped. Defaults to false. */
+  public keepAfterStop: boolean;
+  /** A comma-separated list of accounts to unlock when starting the node. */
+  public unlock: string;
+  /**
+   * Path to a password file with one line per unlocked account.
+   * We have to use a password file when unlocking as we cannot provide the password on the command
+   * line when running the docker container as a daemon in the background.
+   */
+  public password: string;
+
+  /**
+   * @param options The options from the command line.
+   */
+  constructor(options: {
+    mosaicDir: string,
+    port: string,
+    rpcPort: string,
+    websocketPort: string,
+    keepAfterStop: boolean,
+    unlock: string,
+    password: string,
+  }) {
+    Object.assign(this, options);
+  }
+
+  /**
+   * Adds the common node options to a command. Does not include unlock and password. Add these
+   * separately if you need them.
+   * @param command The command where to add the options.
+   * @returns The command with the options added.
+   */
+  public static addCliOptions(command): any {
+    command
+      .option('-d,--mosaic-dir <dir>', 'a path to a directory where the chain data will be stored', DEFAULT_DATA_DIR)
+      .option('-p,--port <port>', 'the first port to use for forwarding from host to container', NodeOptions.stringToDecimal, DEFAULT_PORT)
+      .option('-r,--rpc-port <port>', 'the first RPC port to use for forwarding from host to container', NodeOptions.stringToDecimal, DEFAULT_RPC_PORT)
+      .option('-w,--ws-port <port>', 'the first WS port to use for forwarding from host to container', NodeOptions.stringToDecimal, DEFAULT_WS_PORT)
+      .option('-k,--keep', 'if set, the container will not automatically be deleted when stopped');
+
+    return command;
+  }
+
+  /**
+  * Parses the commander options and returns an Options object.
+  * @param options Options as they are given by commander.
+  * @returns The parsed options with defaults for options that are missing from the command line.
+  */
+  public static parseOptions(options): NodeOptions {
+    const parsedOptions: NodeOptions = new NodeOptions({
+      mosaicDir: options.mosaicDir || DEFAULT_DATA_DIR,
+      port: options.port || DEFAULT_PORT,
+      rpcPort: options.rpcPort || DEFAULT_RPC_PORT,
+      websocketPort: options.wsPort || DEFAULT_WS_PORT,
+      keepAfterStop: options.keep ? true : false,
+      unlock: options.unlock || '',
+      password: options.password || '',
+    });
+
+    parsedOptions.mosaicDir = Directory.sanitize(parsedOptions.mosaicDir);
+
+    if (parsedOptions.password !== '') {
+      parsedOptions.password = Directory.sanitize(parsedOptions.password);
+    }
+
+    if (parsedOptions.unlockOrPasswordWithoutTheOther()) {
+      Logger.error('cannot use --unlock or --password without the other');
```

Comment:
We cannot only have one, because:

* password without unlock doesn't make sense
* unlock without password does not work as we don't give the user a `tty` to enter the password on the command line (as it is inside a daemonized docker).

I've been thinking about this a lot, but have not yet found a solution. Ideas?

/cc @sarvesh-ost

Reply:
I see. Basically `--unlock` option contains comma separated account addresses to unlock and password is a file path that contains a passwords for each account in a new line. Nothing should be changed in implemenattion I would say then.